ls1intum/Artemis
# `Programming exercises`: Show input errors on exercise creation aggregated on footer

### Checklist
#### General
#### Client

### Motivation and Context
When creating programming exercises, all input errors will be shown aggregately in vicinity of the 'Generate'-button. This PR is linked to issue #4198. 


### Description
Validation is done in a new method, that validates the input and returns all error messages as an array of type
`{
    translateKey: <translate key as a string for the corresponding error message>,
    translateValues: <a map of values that may be needed for the message>
}` .
All error messages will then be displayed in the red colored badge.

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 Course

1. Log in to Artemis
2. Navigate to Course Administration
3. Navigate to Exercises of any course
4. Navigate to 'New programming exercises'
5. Try all sort of input errors and make sure, that an appropriate error message is shown while hovering over the box left of the 'Generate'-button.
6. Make sure that the error box disappears when all input is valid, the generate button can be clicked and the exercise will be created correctly.

### Review Progress
#### Code Review
#### Manual Tests

### Test Coverage
File: Line - Branch
programming-exercise-update.component.ts: 72.9% - 70.18%; changes made however are covered by 92.9%

### Screenshots
![Bildschirmfoto 2021-11-15 um 21 08 32](https://user-images.githubusercontent.com/73833780/141846986-6a26a718-045b-4c07-b957-678fc265a415.png)
![Bildschirmfoto 2021-11-15 um 21 10 58](https://user-images.githubusercontent.com/73833780/141847220-d08c3bb2-cb65-48ce-b482-cd12bc69d8c7.png)
